ompi/mca/pml/ucx/pml_ucx.c

```diff
@@ -627,15 +630,15 @@ int mca_pml_ucx_recv(void *buf, size_t count, ompi_datatype_t *datatype, int src
     MCA_COMMON_UCX_PROGRESS_LOOP(ompi_pml_ucx.ucp_worker) {
         status = ucp_request_test(req, &info);
         if (status != UCS_INPROGRESS) {
-            mca_pml_ucx_set_recv_status_safe(mpi_status, status, &info);
+            mca_pml_ucx_set_recv_status(mpi_status, status, &info);
```

Comment:
maybe add return value for mca_pml_ucx_set_recv_status(), instead of extra branch in lines 615-616?

Reply:
branch on 615-616 needed to call mca_pml_ucx_set_recv_status. I can move it in to mca_pml_ucx_set_recv_status_safe